### PR TITLE
Development: devcontainer improvements

### PR DESCRIPTION
**Pull request by orcema**
## Proposed change
Serveral enhancements for using devContainres for developpement

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by orcema** [comment]
Do you agree to add documentation for the devContainer at the end of the documentation for development section [https://docs.paperless-ngx.com/development/#:~:text=Making%20custom%20parsers-,Development,-This%20section%20describes](https://docs.paperless-ngx.com/development/#:~:text=Making%20custom%20parsers-,Development,-This%20section%20describes)
and add a line at the top of the page with a link to the bottom of the page to the section for the devContainer ?
image

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -148,10 +148,9 @@ RUN --mount=type=cache,target=/root/.cache/pip/,id=pip-cache \
   set -eux \
   && echo "Installing build system packages" \
     && apt-get update \
-    && apt-get install --yes --quiet ${BUILD_PACKAGES}
-
-RUN set -eux \
-  && npm update npm -g
+    && apt-get install --yes --quiet ${BUILD_PACKAGES} \
+    && npm update npm -g \
+    && npm install -g @angular/cli@~18.2.11
```
Comment (shamoon) [inline .devcontainer/Dockerfile:153]: This doesn't look correct, npm install is not "installing build system packages". Also pinning this to a specific version of the angular cli seems very annoying

File: .devcontainer/vscode/tasks.json [hunk 2]
```diff
@@ -37,100 +85,104 @@
...
-	  },
-	  {
-		"label": "Celery Worker",
-		"type": "shell",
-		"command": "pipenv run celery --app paperless worker -l DEBUG",
-		"group": {
-		  "kind": "build",
-		  "isDefault": true
+		{
+			"label": "Maintenance: Install Angular Frontend for Debug",
```
Comment (shamoon) [inline .devcontainer/vscode/tasks.json:162]:
```suggestion
			"label": "Maintenance: Install Frontend Dependencies",
```

**Review by shamoon — CHANGES_REQUESTED** [review]
Overall Im a little unclear who all of these changes are for as no one has asked for them but I dont know how many people are using the dev container so they're basically fine as long as they are self-contained.

You can add a note to the development docs, but no, please dont make it the second line of the entire page.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (orcema) [reply .devcontainer/Dockerfile:153]: removed with this commit "remove angular cli installation from Dockerfile"

File: .devcontainer/vscode/tasks.json (hunk 2 above)
Comment (orcema) [reply .devcontainer/vscode/tasks.json]: updated with commit "task for documentation build"

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (shamoon) [reply .devcontainer/Dockerfile]: Thanks. why the changes here at all?

Comment (orcema) [reply .devcontainer/Dockerfile]: For new contributors to this project I think it's a kind of easyier to start with devcontainers. All you have to do it's simply open the cloned project in vscode and wait for the dev environement to be ready for debugging purposes. This avoids potential missing packages to be installed on you host system. 
Running `pipenv install --dev` i get the errors here below where as with devcontainer i do not have worry about host issues :-)

image

Comment (shamoon) [reply .devcontainer/Dockerfile]: No sorry I just meant why the changes to these lines in the dockerfile

Comment (orcema) [reply .devcontainer/Dockerfile]: initally i thaught i should install the angular cli within the Dockerfile and finaly your comment made me realise that indeed the angluar cli has not to be installed by the Dockerfile

File: .devcontainer/Dockerfile
Comment (shamoon) [inline .devcontainer/Dockerfile:1]: Perhaps it will be clearer if I comment on the current changes, what is the purpose of these changes to the Dockerfile, combining the two RUN commands? Seems like this change is unnecessary now

**Comment by orcema** [comment]
ok i'm done with devcontainer enhancements as well as their documentation.

File: .devcontainer/Dockerfile
Comment (orcema) [reply .devcontainer/Dockerfile:1]: ok

Comment (orcema) [reply .devcontainer/Dockerfile:1]: reverted

File: docs/development.md [hunk 3]
```diff
@@ -450,3 +450,66 @@ def myparser_consumer_declaration(sender, **kwargs):
...
+
+#### 1) Clone the repository on your machine
+
+#### 2) Open the paperless-ngx folder in Visual Studio Code
+
+#### 3) Visual Studio Code will prompt you with "Reopen in container"
+
+![alt text](assets/screenshots/vscode_prompt_devcontainer.png){: style="width:32%"}
+
+#### 4) Wait for the dev environment to spinn up
```
Comment (tooomm) [inline docs/development.md:475]:
```suggestion
#### 4) Wait for the dev environment to spin up
```

File: docs/development.md [hunk 4]
```diff
@@ -450,3 +450,66 @@ def myparser_consumer_declaration(sender, **kwargs):
...
+
+This task will:
+
+-   initialise the database tables
+-   create the superuser for paperless
+
+    > you will be prompted to create the superuser for paperless
+
+-   compile the front end for production
+    > this is necessary if run run only the backend server in order to have a front end application
```
Comment (tooomm) [inline docs/development.md:501]:
```suggestion
    > this is necessary if you run only the backend server in order to have a front end application
```

File: docs/development.md [hunk 5]
```diff
@@ -450,3 +450,66 @@ def myparser_consumer_declaration(sender, **kwargs):
...
+
+-   Start either run the fullstack debug or individual debug processes
+
+    ![alt text](assets/screenshots/devcontainer_run_debug.png){: style="width:32%"}
+
+> note that the debug option **Chrome: Debug Angular Frontend** is a 2 step process.
+>
+> first it starts task labeled **Start: Frontend Angular** and when angular is ready it will open a chrome browser for debugging
+
+**Note**: If you only want to spinn up the whole project without debugging you can run the task labeled **Project Start: Run all Services**
```
Comment (tooomm) [inline docs/development.md:515]:
```suggestion
**Note**: If you only want to spin up the whole project without debugging you can run the task labeled **Project Start: Run all Services**
```

**Review by tooomm — COMMENTED** [review]
Few tiny typos

File: docs/development.md (hunk 5 above)
Comment (orcema) [reply docs/development.md]: typos fixed

**Review by shamoon — APPROVED** [review]
I have simplified the docs. We're not writing a VS Code manual here and especially given the target audience for this, we certainly dont need that much info nor screenshots.

Thanks for the PR

**Comment by StefanSchoof** [comment]
There now two places with devcontainer docs:
 - https://github.com/paperless-ngx/paperless-ngx/blob/dev/.devcontainer/README.md
 - https://github.com/paperless-ngx/paperless-ngx/blob/dev/docs/development.md#using-visual-studio-code-devcontainer

This PR changed some of the task that are used in the Devcontainer readme. (I must say I find the place of the readme in the Devcontainer folder a little bit surprising)

**Comment by shamoon** [comment]
Ha, funny timing on the comment there, but fair enough. I will update that

**Comment by orcema** [comment]
> There now two places with devcontainer docs:
> 
> * https://github.com/paperless-ngx/paperless-ngx/blob/dev/.devcontainer/README.md
> * https://github.com/paperless-ngx/paperless-ngx/blob/dev/docs/development.md#using-visual-studio-code-devcontainer
> 
> This PR changed some of the task that are used in the Devcontainer readme. (I must say I find the place of the readme in the Devcontainer folder a little bit surprising)

I must admit i didn't notice the readme.md in the .devcontainer folder. I mainly focused on the documentation in the folder "docs". Wouldn't it be better to move this readme in the folder "docs" too ?
